evalml/pipelines/components/transformers/encoders/ordinal_encoder.py

```diff
@@ -0,0 +1,320 @@
+            categories = "auto"
+        elif self.parameters["categories"] is not None:
+            # Categories specified - make sure they match the ordinal columns
+            input_categories = self.parameters["categories"]
+            if len(input_categories) != len(self.features_to_encode) or not isinstance(
+                input_categories[0],
+                list,
+            ):
+                raise ValueError(
+                    "Categories argument must contain a list of categories for each ordinal feature",
+                )
+
+            # Categories, as they're passed into SKOrdinalEncoder should be in the same order
+            # as the data's Ordinal.order categories even if it's a subset
+            categories = []
+            for i, col_categories in enumerate(input_categories):
+                categories_order = ww_logical_types[self.features_to_encode[i]].order
+
+                ordered_categories = [
+                    cat for cat in categories_order if cat in col_categories
+                ]
+                categories.append(ordered_categories)
+        else:
+            # Categories unspecified - use ordered categories from a columns' Ordinal logical type
+            # and account for top_n
+            categories = []
+            for col in X_t[self.features_to_encode]:
+                ltype = ww_logical_types[col]
+                # Copy the order list, since we might mutate it later by adding nans
+                # and don't want to impact the Woodwork types
+                column_ordered_categories = ltype.order.copy()
+
+                if top_n is None or len(column_ordered_categories) <= top_n:
+                    unique_values = column_ordered_categories
+                else:
+                    # Don't count nans as a value
+                    value_counts = X_t[col].value_counts(dropna=True).to_frame()
+                    value_counts = value_counts.sample(
+                        frac=1,
+                        random_state=self._initial_state,
+                    )
+                    value_counts = value_counts.sort_values(
+                        [col],
+                        ascending=False,
+                        kind="mergesort",
+                    )
+                    unique_values = value_counts.head(top_n).index.tolist()
+                    # Categories should be in the same order as the data's Ordinal.order categories
+                    # instead of sorted by most frequent
+                    unique_values = [
+                        cat for cat in column_ordered_categories if cat in unique_values
+                    ]
+
+                categories.append(unique_values)
+
+        # Add any null values into the categories lists so that they aren't treated as unknown values
+        # This is needed because Ordinal.order won't indicate if nulls are present, and SKOrdinalEncoder
```

Comment:
It's not specifically mentioned in any of the docs, from what I can tell. But I do think it's intended and necessary for two reasons:
1. The Ordinal ltype sets the pandas `category` dtype's categories to be `order`, and the categories are not allowed to be null, or you get `ValueError: Categorical categories cannot be null` from pandas.
2. The implementation of Ordinal, when validating the `order`, checks for values in the series that are not in `order` by [explicitly excluding nans from that check](https://github.com/alteryx/woodwork/blob/main/woodwork/logical_types.py#L638). This allows nans to be in the series without being in the `order`, without which, you couldn't have nans in an Ordinal column at all (because of reason 1).

So maybe I just make a documentation issue asking them to add a note to the logical type docstring about how nans are handled

Reply:
Gotcha, a documentation update would definitely be helpful!

Reply:
Created a woodwork issue: https://github.com/alteryx/woodwork/issues/1538